eng/pipelines/templates/stages/archetype-spec-gen-sdk.yml

```diff
@@ -243,20 +248,28 @@ extends:
                 WorkingDirectory: $(SdkRepoDirectory)
                 ScriptDirectory: $(SdkRepoDirectory)/eng/common/scripts
 
-            - task: PowerShell@2
-              displayName: Create pull request
-              condition: and(succeeded(), eq(variables['HasChanges'], 'true'), ne(variables['Build.Reason'], 'PullRequest'))
-              inputs:
-                pwsh: true
-                workingDirectory: $(SdkRepoDirectory)
-                filePath: $(SdkRepoDirectory)/eng/common/scripts/Submit-PullRequest.ps1
-                arguments: >
-                  -RepoOwner "$(SdkRepoOwner)"
-                  -RepoName "$(SdkRepoName)"
-                  -BaseBranch "main"
-                  -PROwner "azure-sdk"
-                  -PRBranch "$(PrBranch)-$(Build.BuildId)"
-                  -AuthToken "$(azuresdk-github-pat)"
-                  -PRTitle "$(PrTitle)-generated-from-$(Build.DefinitionName)-$(Build.BuildId)"
-                  -PRBody "$(GeneratedSDKInformation)"
-                  -OpenAsDraft $true
+            - ${{ if not(endsWith(variables['SdkRepoName'], '-pr')) }}:
+              - task: PowerShell@2
+                displayName: Create pull request
+                condition: and(succeeded(), eq(variables['HasChanges'], 'true'), ne(variables['Build.Reason'], 'PullRequest'))
+                inputs:
+                  pwsh: true
+                  workingDirectory: $(SdkRepoDirectory)
+                  filePath: $(SdkRepoDirectory)/eng/common/scripts/Submit-PullRequest.ps1
+                  arguments: >
+                    -RepoOwner "$(SdkRepoOwner)"
+                    -RepoName "$(SdkRepoName)"
+                    -BaseBranch "main"
+                    -PROwner "azure-sdk"
+                    -PRBranch "$(PrBranch)-$(Build.BuildId)"
+                    -AuthToken "$(azuresdk-github-pat)"
+                    -PRTitle "$(PrTitle)-generated-from-$(Build.DefinitionName)-$(Build.BuildId)"
+                    -PRBody "$(GeneratedSDKInformation)"
+                    -OpenAsDraft $true
+
+          - ${{ if eq(variables['Build.Reason'], 'Manual') }}:
```

Comment:
We only want to include this step in case of `pull request trigger`.

Reply:
Oh, sorry. I did this so I can test it out. I will update it